src/payment/models/dex/strategies/purchase-liquidity/purchase-liquidity.facade.ts

```diff
@@ -0,0 +1,93 @@
+import { DeFiChainCryptoStrategy } from './impl/defichain-crypto.strategy';
+import { EthereumCryptoStrategy } from './impl/ethereum-crypto.strategy';
+import { PurchaseLiquidityStrategy } from './impl/base/purchase-liquidity.strategy';
+import { DeFiChainPoolPairStrategy } from './impl/defichain-poolpair.strategy';
+import { DeFiChainStockStrategy } from './impl/defichain-stock.strategy';
+import { BscTokenStrategy } from './impl/bsc-token.strategy';
+import { BitcoinStrategy } from './impl/bitcoin.strategy';
+import { EthereumTokenStrategy } from './impl/ethereum-token.strategy';
+
+enum Alias {
+  BITCOIN = 'Bitcoin',
+  BSC_CRYPTO = 'BscCrypto',
+  BSC_TOKEN = 'BscToken',
+  DEFICHAIN_POOL_PAIR = 'DeFiChainPoolPair',
+  DEFICHAIN_STOCK = 'DeFiChainStock',
+  DEFICHAIN_CRYPTO = 'DeFiChainCrypto',
+  ETHEREUM_CRYPTO = 'EthereumCrypto',
+  ETHEREUM_TOKEN = 'EthereumToken',
+}
+
+export { Alias as PurchaseLiquidityStrategyAlias };
+
+@Injectable()
+export class PurchaseLiquidityStrategies {
+  protected readonly strategies = new Map<Alias, PurchaseLiquidityStrategy>();
+
+  constructor(
+    bitcoin: BitcoinStrategy,
+    bscCrypto: BscCryptoStrategy,
+    bscToken: BscTokenStrategy,
+    deFiChainCrypto: DeFiChainCryptoStrategy,
+    @Inject(forwardRef(() => DeFiChainPoolPairStrategy))
+    deFiChainPoolPair: DeFiChainPoolPairStrategy,
+    deFiChainStock: DeFiChainStockStrategy,
+    ethereumCrypto: EthereumCryptoStrategy,
+    ethereumToken: EthereumTokenStrategy,
+  ) {
+    this.strategies.set(Alias.BITCOIN, bitcoin);
+    this.strategies.set(Alias.BSC_CRYPTO, bscCrypto);
+    this.strategies.set(Alias.BSC_TOKEN, bscToken);
+    this.strategies.set(Alias.DEFICHAIN_POOL_PAIR, deFiChainPoolPair);
+    this.strategies.set(Alias.DEFICHAIN_STOCK, deFiChainStock);
+    this.strategies.set(Alias.DEFICHAIN_CRYPTO, deFiChainCrypto);
+    this.strategies.set(Alias.ETHEREUM_CRYPTO, ethereumCrypto);
+    this.strategies.set(Alias.ETHEREUM_TOKEN, ethereumToken);
+  }
+
+  getPurchaseLiquidityStrategy(criteria: Asset | Alias): PurchaseLiquidityStrategy {
+    return criteria instanceof Asset ? this.getByAsset(criteria) : this.getByAlias(criteria);
+  }
+
+  //*** HELPER METHODS ***//
+
+  private getByAlias(alias: Alias): PurchaseLiquidityStrategy {
+    const strategy = this.strategies.get(alias);
+
+    if (!strategy) throw new Error(`No PurchaseLiquidityStrategy found. Alias: ${alias}`);
+
+    return strategy;
+  }
+
+  private getByAsset(asset: Asset): PurchaseLiquidityStrategy {
+    const alias = this.getAlias(asset);
+
+    return this.getByAlias(alias);
+  }
+
+  private getAlias(asset: Asset): Alias {
+    const { blockchain, category: assetCategory } = asset;
+
+    if (blockchain === Blockchain.BITCOIN) return Alias.BITCOIN;
+
+    if (blockchain === Blockchain.BINANCE_SMART_CHAIN) {
+      if (assetCategory === AssetCategory.CRYPTO) return Alias.BSC_CRYPTO;
+      if (assetCategory === AssetCategory.STOCK) return Alias.BSC_TOKEN;
+    }
```

Comment:
Perhaps we need to discuss this. From what I saw, BSC_CRYPTO is for native BNB and BSC_TOKEN for all BEP-20 tokens, is this right? Then the differentiation based on AssetCategory is not really fitting 🤔 
Same for other facades and Ethereum blockchain.

Reply:
It seems to be very similar for all chains. There are the native asset and multiple wrapped tokens, which need to be handled differently.